rust/garbage_collector/src/garbage_collector_orchestrator_v2.rs

```diff
@@ -213,6 +215,55 @@ impl Orchestrator for GarbageCollectorOrchestrator {
 }
 
 impl GarbageCollectorOrchestrator {
+    async fn get_eligible_soft_deleted_collections_to_gc(
+        &mut self,
+        all_collection_ids: Vec<CollectionUuid>,
+    ) -> Result<HashSet<CollectionUuid>, GarbageCollectorError> {
+        let soft_delete_statuses = self
+            .sysdb_client
+            .batch_get_collection_soft_delete_status(all_collection_ids.clone())
+            .await
+            .map_err(|e| GarbageCollectorError::SysDbMethodFailed(e.to_string()))?;
+
+        let all_collection_ids: HashSet<CollectionUuid> = all_collection_ids.into_iter().collect();
+
+        let soft_deleted_collections_to_gc = soft_delete_statuses
+            .iter()
+            .filter_map(|(collection_id, status)| {
+                if *status && all_collection_ids.contains(collection_id) {
+                    Some(*collection_id)
+                } else {
+                    None
+                }
+            })
+            .collect::<Vec<_>>();
+
+        let collections = self
+            .sysdb_client
+            .get_collections(GetCollectionsOptions {
```

Comment:
why can't you directly use this for `all_collection_ids`? why do you need a separate `batch_get_collection_soft_delete_status`?

Reply:
https://github.com/chroma-core/chroma/pull/4719/files/e81e954f928ec18a7c144e4c4607939657d1500b#r2125184184

we would need to add `is_deleted` to the Collection struct which seems a little strange